feat(types): add support for custom types

adrianschmidt opened the pull request:
Inspired by #104, but slightly different implementation that doesn't require specifying all types just to add one or more custom ones.

The limitation in this implementation is that, while types defined in **conventional-commit-types** can be overridden, they cannot be removed entirely.

fix #93

adrianschmidt commented:
I realise this probably needs tests before it's accepted, but after looking at the test file, I unfortunately feel I that I don't have the time to do that right now. But since I forked the repo to add this functionality for our own use, I thought I could just as well open a PR and see what you thought.

If the addition itself is welcomed, and only the lack of tests stops it from being merged, please tell me so, and I'll try to find the time to add those tests!

jimthedev commented:
Would be happy to merge if you want to rebase. There's no reason to not have both, correct? If someone wants to swap out all types that is type. If they want to add them then us additionalTypes?

adrianschmidt commented:
Sorry, pushed a commit that shouldn't be there. I'll remove it.

adrianschmidt commented:
There we go. Rebased and ready. I also added a second commit updating the readme regarding `types`, so the documenation for `types` and `additionalTypes` match, and hopefully is a little clearer. Please tell me if I should remove it, or make any changes!

adrianschmidt commented:
@jimthedev Not sure you got a notification for my last message, so I thought I'd ping you 🙂

ThisIsMissEm commented:
@adrianschmidt this seems like a change that'd be better upstream in cz-cli? Allow loading a .js file there, and then you can programmatically manipulate the types object & require conventional-commit-types.

adrianschmidt commented:
> @adrianschmidt this seems like a change that'd be better upstream in cz-cli? Allow loading a .js file there, and then you can programmatically manipulate the types object & require conventional-commit-types.

Hmm… yeah, that might be true. Sorry for not getting back to this earlier, I haven't had the peace of mind to sit down and look at it lately. I'll close this for now, and might come back with something for cz-cli later on.

Thanks for your help!
